app/helpers/application_helper.rb

```diff
@@ -1,2 +1,23 @@
 module ApplicationHelper
+    class CodeRayify < Redcarpet::Render::Safe
+        # uses html_escape flag to prevent html/xss injections
+        def block_code(code, language)
+            CodeRay.scan(code, language || :text).div
+        end
+    end
+
+    def markdown(content)
```

Comment:
This crashes when content is nil instead of a string. In this case, when a Node is a `topic`-type Node, content body can be nil.

Reply:
Do you have an example of this? I can't seem to reproduce it. What type of error do you get?

Reply:
```
Started GET "/5" for 127.0.0.1 at 2021-02-03 11:27:29 +1100
Processing by NodesController#show as HTML
  Parameters: {"id"=>"5"}
  User Load (1.7ms)  SELECT "users".* FROM "users" WHERE "users"."id" = $1 ORDER BY "users"."id" ASC LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ app/controllers/nodes_controller.rb:78:in `set_user'
  UsersGroup Load (1.4ms)  SELECT "users_groups".* FROM "users_groups" WHERE "users_groups"."user_id" = $1 LIMIT $2  [["user_id", 1], ["LIMIT", 1]]
  ↳ app/models/user.rb:34:in `groups'
  SQL (3.0ms)  SELECT "nodes"."id" AS t0_r0, "nodes"."author_id" AS t0_r1, "nodes"."parent_id" AS t0_r2, "nodes"."depth" AS t0_r3, "nodes"."n_children" AS t0_r4, "nodes"."n_descendants" AS t0_r5, "nodes"."created_at" AS t0_r6, "nodes"."updated_at" AS t0_r7, "content_versions"."id" AS t1_r0, "content_versions"."author_id" AS t1_r1, "content_versions"."node_id" AS t1_r2, "content_versions"."title" AS t1_r3, "content_versions"."body" AS t1_r4, "content_versions"."created_at" AS t1_r5, "content_versions"."updated_at" AS t1_r6, "authors"."id" AS t2_r0, "authors"."name" AS t2_r1, "authors"."user_id" AS t2_r2, "authors"."public" AS t2_r3, "authors"."created_at" AS t2_r4, "authors"."updated_at" AS t2_r5, "users"."id" AS t3_r0, "users"."username" AS t3_r1, "users"."created_at" AS t3_r2, "users"."updated_at" AS t3_r3, "users"."email" AS t3_r4, "users"."encrypted_password" AS t3_r5, "users"."reset_password_token" AS t3_r6, "users"."reset_password_sent_at" AS t3_r7, "users"."remember_created_at" AS t3_r8 FROM "nodes" INNER JOIN "node_inherited_authz_reads" ON "node_inherited_authz_reads"."node_id" = "nodes"."id" LEFT OUTER JOIN "content_versions" ON "content_versions"."node_id" = "nodes"."id" LEFT OUTER JOIN "authors" ON "authors"."id" = "content_versions"."author_id" LEFT OUTER JOIN "users" ON "users"."id" = "authors"."user_id" INNER JOIN (WITH RECURSIVE "nodes__recursive" AS ( SELECT "nodes"."id", "nodes"."parent_id" FROM "nodes" WHERE "nodes"."id" = $1 UNION ALL SELECT "nodes"."id", "nodes"."parent_id" FROM "nodes" INNER JOIN "nodes__recursive" ON "nodes__recursive"."id" = "nodes"."parent_id" ) SELECT "nodes__recursive".* FROM "nodes__recursive") AS "nodes__recursive" ON "nodes"."id" = "nodes__recursive"."id" WHERE "node_inherited_authz_reads"."groups" && $2 LIMIT $3  [["id", 5], ["groups", "{admins,subscribers,all}"], ["LIMIT", 1000]]
  ↳ app/models/node.rb:129:in `sort'
  TagDecl Load (1.5ms)  SELECT "tag_decls".* FROM "tag_decls" WHERE "tag_decls"."anchored_type" = $1 AND "tag_decls"."tag" = $2 AND "tag_decls"."user_id" = $3 AND "tag_decls"."target_type" = $4 AND "tag_decls"."anchored_id" IN ($5, $6)  [["anchored_type", "Node"], ["tag", "view"], ["user_id", nil], ["target_type", "UserTag"], [nil, 5], [nil, 88519]]
  ↳ app/models/node.rb:129:in `sort'
  Rendering layout layouts/application.html.erb
  Rendering nodes/show.html.erb within layouts/application
  Rendered nodes/_subtree_draw.html.erb (Duration: 0.3ms | Allocations: 103)
  Rendered nodes/_subtree_draw.html.erb (Duration: 1.0ms | Allocations: 270)
  Rendered nodes/_subtree_full.html.erb (Duration: 1.3ms | Allocations: 370)
  Rendered nodes/_view_comment_inline.html.erb (Duration: 480774.2ms | Allocations: 42266)
  Rendered nodes/_view_topic.html.erb (Duration: 495862.2ms | Allocations: 50476)
  Rendered nodes/_view_node.html.erb (Duration: 495862.9ms | Allocations: 50683)
  Rendered nodes/show.html.erb within layouts/application (Duration: 495863.3ms | Allocations: 50835)
  Rendered layout layouts/application.html.erb (Duration: 495863.5ms | Allocations: 50906)
Completed 500 Internal Server Error in 495886ms (ActiveRecord: 7.7ms | Allocations: 58248)


  
ActionView::Template::Error (wrong argument type nil (expected String)):
     7:   <% end %>
     8:   <div class='comment-block'>
     9:     <div class="content">
    10:       <%= markdown(node.content.body) %>
    11:     </div>
    12:   </div>
    13:   <div class='comment-block is-small'>
  
app/helpers/application_helper.rb:21:in `render'
app/helpers/application_helper.rb:21:in `markdown'
app/views/nodes/_view_comment_inline.html.erb:10
app/views/nodes/_view_topic.html.erb:55
app/views/nodes/_view_topic.html.erb:50:in `each'
app/views/nodes/_view_topic.html.erb:50
app/views/nodes/_view_node.html.erb:2
app/views/nodes/show.html.erb:7
```

Reply:
Please ignore the long render time, I had a debugger attached which delayed the return.

Reply:
Thanks for the spot, I've added a new commit that fixes it.